crates/ide_completion/src/context.rs

```diff
@@ -29,20 +29,25 @@ pub(crate) enum PatternRefutability {
     Irrefutable,
 }
 
+#[derive(Debug)]
+pub(super) enum PathKind {
+    Expr,
```

Comment:
`Value` might be a better name, since this includes patterns?

Reply:
This doesn't include patterns as it checks for `PathExpr`, which cannot appear in pattern position. Patterns will cause the `kind` field to just fall back to `None` for the time being.

Reply:
Well ok maybe it _should_ include patterns? :smile:

Reply:
It *could* but there is no use for it *yet* so I don't think there is point in adding it yet. I'll tackle pattern things again next probably so I'll see if there is a use for it then.